### DISCUSSION TITLE
Update description list component

### DISCUSSION THREAD

**Discussion by tonge495** [twilio-labs/paste#3285] [Ideas] [answered]
Hi all,

Hope you are doing well!

I have run into a few scenarios recently for various projects where I had to use the description list component (see screenshots below):
![Screenshot 2023-06-20 at 3 04 54 PM](https://github.com/twilio-labs/paste/assets/9394413/e0320e83-4943-491b-a576-c82417dfe363)
![Screenshot 2023-06-20 at 3 05 48 PM](https://github.com/twilio-labs/paste/assets/9394413/77f98b59-229d-4847-9cc3-5dd8a954c22c)
![Screenshot 2023-06-20 at 3 05 53 PM](https://github.com/twilio-labs/paste/assets/9394413/175514bb-b456-482c-80ea-86c71b875ef7)

Just wondering, is there an opportunity for the Paste team to update the component moving forward so that it also accommodates for a horizontal layout?  The vertical layout has not always worked for every use case and in some cases, it is helpful to have a horizontal layout.

cc: @HartiganHM

**Comment by serifluous** [comment]
Hey @tonge495 ! Can you share what the intended design you're proposing is? Are you looking to have your team build what you've shared in the screenshot, or something else?

**Reply by tonge495** [reply]
Here is the full screenshots if that helps. Both are more less finalized and in the development phase:
The first one is essentially a view of information related to an Advisory (or a customer communication/email)
![Advisories Details Complete](https://github.com/twilio-labs/paste/assets/9394413/7d1edf21-c767-4459-a9b0-7448c899f273)

The second is a view of support ticket details:
<img width="1496" alt="One Twilio - Ticket details no CCs entered" src="https://github.com/twilio-labs/paste/assets/9394413/60f6ac73-15cf-4438-83be-c8a573f52cfe">

We were originally advised to use the description list component for the areas where there are small snippets of information but the component does not include an option to have a horizontal layout.  Hope that helps a bit.

**Reply by serifluous** [reply]
Sorry, need to clarify my question more! When you refer to a "horizontal layout", is that:

1. what you're showing in your screenshots (like, multiple description lists horizontally next to each other), or
2. not what you're showing in your screenshots, and you mean one description list with the Description List Term and Description Details in one single row?

**Reply by tonge495** [reply]
Ah sorry I misunderstood your question!  It's just one description list with the Description List Term and Description Details in one single row.

**Reply by serifluous** [reply]
Ah got it. We didn't build a horizontal version of Description List because it makes it difficult to scan for term + detail pairings, especially when you have multiple description lists in a column.

They also tend to be used mixed in with form fields in Twilio products, which are vertically aligned, too.

What issues do you run into with the vertical layout, or where/how would a horizontal layout help out?

**Reply by tonge495** [reply]
We are using the description list to display snippets of important summary information for the user.  With the current scenarios that we have, the horizontal is preferable because it provides this information all in one row, rather than the user having to potentially scroll down in order to see all the information before delving into more details.

**Reply by serifluous** [reply]
Would you mind mocking up an example? I'm not sure which description lists in your screenshots would help save on vertical space if they were horizontally aligned.

**Reply by tonge495** [reply]
This is more about providing the summary data at the top of the page so that a user can read summary data from the top left to right of the page before delving into more details.  I believe OneAdmin is using a somewhat similar pattern in the Account and Billing sections:

<img width="812" alt="Screenshot 2023-06-22 at 1 03 39 PM" src="https://github.com/twilio-labs/paste/assets/9394413/c4a4e2b3-3f11-411c-88b0-f8b7be27e6f1">

Happy to schedule a time to chat on this further if needed but also totally understand if it's not a huge priority for Paste right now. 🙂

**Reply by serifluous** [reply]
Sure, could you sign up for office hours? https://docs.google.com/spreadsheets/d/10vD91zdwr9hjgDOV3HSGnoC0LgvYw-thmn3X_kvW0xk/edit#gid=664218049

I'm not seeing a horizontally-aligned description list in the One Admin design you shared, so I think I might still be misunderstanding.

**Reply by serifluous** [reply]
Sorry, I think I know where I was getting confused. Is this what you were looking for? https://codesandbox.io/s/youthful-volhard-fvmm9z?file=/src/index.tsx

**Reply by tonge495** [reply]
Yes that's right!  Did you have any thoughts around potentially offering this for the Description List component?

**Reply by serifluous** [reply, marked as answer]
It's doable with the component right now by wrapping the DescriptionListSets with the [horizontal Stack component](https://paste.twilio.design/components/stack) like I shared in the codesandbox.

Or by using Grid if you wanted to align all the Description Lists you have on a page into uniform columns (which we'd recommend): https://paste.twilio.design/components/grid

We likely won't directly build a horizontal variant into Description List since it's a little more straightforward to do with Stack or Grid. It could be something we add to our Figma library, though, if that's what you're looking for, especially since it's part of our Object Details pattern: https://paste.twilio.design/patterns/object-details

**Reply by tonge495** [reply]
Got it thank you for the resources!

That would be great if you are able to add it to the Figma library eventually, especially since I think we may be using it again in the future.  If the code is not going to be updated, would it be possible to include what you shared on the Paste documentation (https://paste.twilio.design/components/description-list)?

**Reply by serifluous** [reply]
Yep, ticketed the work to add it to Figma here: https://issues.corp.twilio.com/browse/DSYS-2871

Description List also hasn't gone through a peer review yet, so there are definitely improvements overall we could make to the docs. We'll take it into account when we get to the peer review. Thanks!

**Reply by tonge495** [reply]
Awesome, much appreciated!

Also, sorry if I wasn't super clear in my initial responses, remote communications can be a little confusing sometimes.  😅 Glad we were able to get things squared away though!

**Reply by serifluous** [reply]
No worries! Remote communication is definitely tough. I got a little confusing on my end for sure and was conflating Description List & Description List Set.
